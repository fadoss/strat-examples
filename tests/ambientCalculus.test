semantics/ambients/ambients-semantics

srew M using norep .
srew N using norep .
srew R using norep .
srew firewall(stop, stop) using cardellideco(4) .
srew [1] fail(stop, stop) using ecd(3) .
continue 1 .
