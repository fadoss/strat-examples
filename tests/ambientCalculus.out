==========================================
srewrite in MAIN : M using norep .

Solution 1
rewrites: 12
result NSProcess: 'n{0}[in['n{0}] . stop | 'n{0}[stop]]

No more solutions.
rewrites: 12
==========================================
srewrite in MAIN : N using norep .

Solution 1
rewrites: 23
result NSProcess: 'm{0}['n{0}[in['n{0}] . stop | 'n{0}[stop]]]

No more solutions.
rewrites: 23
==========================================
srewrite in MAIN : R using norep .

No solution.
rewrites: 12
==========================================
srewrite in MAIN : firewall(stop, stop) using cardellideco(4) .

Solution 1
rewrites: 148
result NSProcess: new['k : W]new['m : W]('m{0}['n{0}[stop]])

No more solutions.
rewrites: 148
==========================================
srewrite [1] in MAIN : fail(stop, stop) using ecd(3) .

Solution 1
rewrites: 251
result Process: err1

Solution 2
rewrites: 55
result Process: stop
Bye.
